table/block_fetcher.cc

```diff
@@ -103,8 +104,13 @@ bool BlockFetcher::TryGetCompressedBlockFromPersistentCache() {
   if (cache_options_.persistent_cache &&
       cache_options_.persistent_cache->IsCompressed()) {
     // lookup uncompressed cache mode p-cache
+    std::unique_ptr<char[]> raw_data;
+    raw_data.reset(heap_buf_.release());
```

Comment:
What if `heap_buf_` is allocated with the custom allocator? (with a custom deleter) 

`LookupRawPage` may reset `*raw_data` to something else (see, for example, `BlockCacheTier::Lookup`, which does `val->reset(new char[blk_val.size()]);`), which will try to delete `raw_data` (formerly `heap_buf_`) with `delete[]`, not with the custom deleter.

Reply:
Seems like `LookupRawPage` never *reads* from the passed-in unique_ptr, so how about this instead:

```
unique_ptr<char[]> raw_data;
LookupRawPage(..., &raw_data, ...);
if (raw_data) {
    heap_buf_ = CacheAllocationPtr(raw_data.release());
}
```

Reply:
Yup